src/components/input-date-picker/input-date-picker.tsx

```diff
@@ -912,17 +914,21 @@ export class InputDatePicker
 
   private warnAboutInvalidValue(value: string): void {
     console.warn(
-      `The specified value "${value}" does not conform to the required format, "yyyy-MM-dd".`
+      `The specified value "${value}" does not conform to the required format, "YYYY-MM-DD".`
     );
   }
 
+  private commonDateSeparators = [".", "-", "/"];
+
   private formatNumerals = (value: string): string =>
     value
       ? value
           .split("")
           .map((char: string) =>
-            numberKeys.includes(char)
-              ? numberStringFormatter.numberFormatter.format(Number(char))
+            this.commonDateSeparators?.includes(char)
+              ? this.localeData?.separator
```

Comment:
As a note before I merge, I added this commonDateSeparator code as a convienence in #5488. It used to automatically convert incorrect date separators as you typed, but it shouldn't have affected anything if the separator was typed correctly. 

Here is a demo from `beta.97`, If you type "2020-1-1" it will automatically convert to "2020.1.1". https://codepen.io/benesri/pen/qByZmXd

I removed it in #5887, because at that point the auto conversion no longer worked and it was causing JS Handle errors in the e2e test I added. For some reason removing it caused "Issue 1" in #5969, so this PR adds it back.

But at some point we should probably figure out why the auto conversion is no longer working and why the date separator doesn't work without this code, which wasn't meant for that.

cc @eriklharper

Reply:
I bet the reason the e2e environment had a problem with it could be due to the encoding on the characters, but yeah this is a good item to research further.  Just an idea, but one way to set `commonDateSeparators` are to get the separator chars from the locales themselves using the Intl.DateTimeFormat.formatToParts api that way we can ensure the correct character is being used.

Reply:
The separator being used is coming from `localeData`, so that will already always be correct. But I'll put in a PR in a sec to make sure all date separators get converted since the code relies on it now lol